File: CS/GrowthWare.Framework/Model/Profiles/MFunctionTypeProfile.cs
using GrowthWare.Framework.Model.Profiles.Base;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GrowthWare.Framework.Model.Profiles
{
    /// <summary>
    /// Class MFunctionTypeProfile
    /// </summary>
    [Serializable(), CLSCompliant(true)]
    public class MFunctionTypeProfile : MProfile
    {
        #region "Member Properties"
        private int m_FunctionTypeSeqId = -1;
        private string m_Description = string.Empty;
        private string m_Template = string.Empty;
        private bool m_IsContent;
        #endregion

        /// <summary>
        /// Will return a Function profile with the default values
        /// </summary>
        /// <remarks></remarks>
        public MFunctionTypeProfile()
        {
        }

        /// <summary>
        /// Will return a fully populated Function type profile.
        /// </summary>
        /// <param name="dataRow">A data row containing the Function type information</param>
        /// <remarks></remarks>
        public MFunctionTypeProfile(DataRow dataRow)
        {
            Initialize(dataRow);
        }

        /// <summary>
        /// Initializes the specified datarow.
        /// </summary>
        /// <param name="Datarow">The datarow.</param>
        protected new void Initialize(DataRow detailRow)
        {
            base.NameColumnName = "NAME";
            base.IdColumnName = "Function_Type_Seq_ID";
            if (detailRow != null) 
            {
                base.Initialize(detailRow);
                m_FunctionTypeSeqId = Id;
                m_Description = base.GetString(detailRow, "DESCRIPTION");
                m_Template = base.GetString(detailRow, "TEMPLATE");
                m_IsContent = base.GetBool(detailRow, "IS_CONTENT");
            }
        }

        /// <summary>
        /// Gets or sets the function_ type_ seq_ ID.
        /// </summary>
        /// <value>The FunctionTypeSeqId.</value>
        public int FunctionTypeSeqId
        {
            get { return m_FunctionTypeSeqId; }
            set { m_FunctionTypeSeqId = value; }
        }

        /// <summary>
        /// Gets or sets the description.
        /// </summary>
        /// <value>The description.</value>
        public string Description
        {
            get { return m_Description; }
            set { if (!string.IsNullOrEmpty(value)) m_Description = value.Trim(); }
        }

        /// <summary>
        /// Gets or sets the TEMPLATE.
        /// </summary>
        /// <value>The TEMPLATE.</value>
        public string Template
        {
            get { return m_Template; }
            set { if(!string.IsNullOrEmpty(value)) m_Template = value.Trim(); }
        }

        /// <summary>
        /// Gets or sets a value indicating whether [I s_ CONTENT].
        /// </summary>
        /// <value><c>true</c> if [I s_ CONTENT]; otherwise, <c>false</c>.</value>
        public bool IsContent
        {
            get { return m_IsContent; }
            set { m_IsContent = value; }
        }
    }
}

File: VB/GrowthWare.Framework/Model/Profiles/MFunctionTypeProfile.vb
Imports GrowthWare.Framework.Model.Profiles.Base
Imports GrowthWare.Framework.Model.Profiles.Interfaces

Namespace Model.Profiles
    <Serializable(), CLSCompliant(True)> _
    Public Class MFunctionTypeProfile
        Inherits MProfile
        Implements IMProfile

#Region "Member Properties"
        Private m_FunctionTypeSeqId = -1
        Private m_Description As String = String.Empty
        Private m_Template As String = String.Empty
        Private m_IsContent As Boolean
#End Region

        ''' <summary>
        ''' Will return a Function profile with the default vaules
        ''' </summary>
        ''' <remarks></remarks>
        Public Sub New()

        End Sub

        ''' <summary>
        ''' Will return a fully populated Function type profile.
        ''' </summary>
        ''' <param name="dataRow">A data row containing the Function type information</param>
        ''' <remarks></remarks>
        Public Sub New(ByVal dataRow As DataRow)
            Initialize(dataRow)
            MyBase.Initialize(dataRow)
        End Sub

        Protected Shadows Sub Initialize(ByVal detailRow As DataRow)
            MyBase.IdColumnName = "Function_Type_Seq_ID"
            MyBase.NameColumnName = "NAME"
            If Not detailRow Is Nothing Then
                MyBase.Initialize(detailRow)
                m_FunctionTypeSeqId = Id
                m_Description = CStr(detailRow("DESCRIPTION"))
                m_Template = CStr(detailRow("TEMPLATE"))
                m_IsContent = CStr(detailRow("IS_CONTENT"))
            End If
        End Sub

        Public Property FunctionTypeSeqId() As Integer
            Get
                Return m_FunctionTypeSeqId
            End Get
            Set(ByVal value As Integer)
                m_FunctionTypeSeqId = value
            End Set
        End Property

        Public Property Description() As String
            Get
                Return m_Description
            End Get
            Set(ByVal Value As String)
                If Not String.IsNullOrEmpty(Value) Then m_Description = Value.Trim()
            End Set
        End Property

        Public Property Template() As String
            Get
                Return m_Template
            End Get
            Set(ByVal value As String)
                If Not String.IsNullOrEmpty(value) Then m_Template = value.Trim()
            End Set
        End Property

        Public Property IsContent() As Boolean
            Get
                Return m_IsContent
            End Get
            Set(ByVal value As Boolean)
                m_IsContent = value
            End Set
        End Property
    End Class
End Namespace